node/impl/ipc/ipc.go

```diff
@@ -100,19 +102,48 @@ func (a *IpcAPI) IpcAddSubnetActor(ctx context.Context, wallet address.Address,
 func (a *IpcAPI) IpcReadGatewayState(ctx context.Context, actor address.Address, tsk types.TipSetKey) (*gateway.State, error) {
 	st := &gateway.State{}
 	if err := a.readActorState(ctx, actor, tsk, st); err != nil {
-		return nil, xerrors.Errorf("error getting gateway actor from StateStore: %w")
+		return nil, xerrors.Errorf("error getting gateway actor from StateStore: %w", err)
 	}
 	return st, nil
 }
 
 func (a *IpcAPI) IpcReadSubnetActorState(ctx context.Context, actor address.Address, tsk types.TipSetKey) (*subnetactor.State, error) {
 	st := &subnetactor.State{}
 	if err := a.readActorState(ctx, actor, tsk, st); err != nil {
-		return nil, xerrors.Errorf("error getting subnet actor from StateStore: %w")
+		return nil, xerrors.Errorf("error getting subnet actor from StateStore: %w", err)
 	}
 	return st, nil
 }
 
+// IpcGetPrevCheckpointForChild gets the latest checkpoint committed for a child subnet.
+// This function is expected to be called in the parent of the checkpoint being populated.
+// It inspects the state in the heaviest block (i.e. latest state available)
+func (a *IpcAPI) IpcGetPrevCheckpointForChild(ctx context.Context, gatewayAddr address.Address, subnet sdk.SubnetID) (cid.Cid, error) {
```

Comment:
Let's rename `IpcAPI` to API. The package is already called `ipc`. In that case we will have clear struct and its methods `func (api *API)`

Reply:
Unfortunately it won't work because the full API expects to receive an anonymous implementation to provide the methods right away and there is already a  `client.API`, so we need some other name :(